common/OPexpect.py

```diff
@@ -93,53 +94,59 @@ def expect(self, pattern, timeout=-1, searchwindowsize=-1, async=False):
 
         state = None
 
-        if r in [1,2,3,4,5,6,7,8,9,10,11,12,13]:
+        if r in [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]:
             # We set the system state to UNKNOWN_BAD as we want to have a path
             # to recover and run the next test, which is going to be to IPL
             # the box again.
             # We do this via a callback rather than any other method as that's
             # just a *lot* easier with current code structure
             if self.failure_callback:
                 state = self.failure_callback(self.failure_callback_data)
-        if r in [1,2,3,4,5,6,7,8]:
+        if r in [1, 2, 3, 4, 5, 6, 7, 8]:
             log = str(self.after)
             l = 0
 
-            while l != 7:
-                l = super(spawn,self).expect(["INFO: rcu_sched self-detected stall on CPU",
-                                              "Watchdog .* Hard LOCKUP",
-                                              "Sending IPI to other CPUs",
-                                              ":mon>",
-                                              "Rebooting in \d+ seconds",
-                                              "Kernel panic - not syncing: Fatal exception",
-                                              "Kernel panic - not syncing: Hard LOCKUP", pexpect.TIMEOUT],
-                                             timeout=15)
+            while l != 8:
+                l = super(spawn, self).expect(["INFO: rcu_sched self-detected stall on CPU",
+                                               "Watchdog .* Hard LOCKUP",
+                                               "Sending IPI to other CPUs",
+                                               ":mon>",
+                                               "Rebooting in \d+ seconds",
+                                               "Kernel panic - not syncing: Fatal exception",
+                                               "Kernel panic - not syncing: Hard LOCKUP",
+                                               "opal_cec_reboot2", pexpect.TIMEOUT],
+                                              timeout=15)
                 log = log + str(self.before) + str(self.after)
-                if l in [2,3,4]:
-                    # We know we have the end of the error message, so let's stop here.
+                if l in [2, 3, 4, 5, 7]:
+                    # We know we have the end of the error message, so let's
+                    # stop here.
                     break
 
             if r == 1 or r == 8:
                 raise KernelSoftLockup(state, log)
             if r == 2:
                 raise KernelBug(state, log)
-            if r == 3:
-                raise KernelPanic(state, log)
             if r == 4 or r == 6 or r == 7:
                 raise KernelHardLockup(state, log)
-            if r == 5 and l == 2:
```

Comment:
I had told the reason offline..sorry did not update this conversation.. previously the code checked for r == 5 which was not correct..  reason I had altered `r == 5` to `r == 3` because when we trigger kdump we get `Kernel panic` and not `Oops: Kernel access of bad area`..  now after this change the code rightly triggers KernelKdump exception when sys crash is triggered with kdump on..